service/service.go

```diff
@@ -2,56 +2,56 @@ package service
 
 import (
 	"context"
-	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+
 	"github.com/ONSdigital/dp-topic-api/api"
 	"github.com/ONSdigital/dp-topic-api/config"
-	"github.com/ONSdigital/go-ns/server"
 	"github.com/ONSdigital/log.go/log"
 	"github.com/gorilla/mux"
 	"github.com/pkg/errors"
 )
 
+// Service contains all the configs, server and clients to run the dp-topic-api API
 type Service struct {
 	Config      *config.Config
-	server      *server.Server
+	Server      HTTPServer
 	Router      *mux.Router
-	API         *api.API
-	HealthCheck *healthcheck.HealthCheck
+	Api         *api.API
+	ServiceList *ExternalServiceList
+	HealthCheck HealthChecker
 }
 
 // Run the service
-func Run(buildTime, gitCommit, version string, svcErrors chan error) (*Service, error) {
-	ctx := context.Background()
+func Run(ctx context.Context, cfg *config.Config, serviceList *ExternalServiceList, buildTime, gitCommit, version string, svcErrors chan error) (*Service, error) {
+
 	log.Event(ctx, "running service", log.INFO)
 
-	cfg, err := config.Get()
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to retrieve service configuration")
-	}
-	log.Event(ctx, "got service configuration", log.Data{"config": cfg}, log.INFO)
+	log.Event(ctx, "using service configuration", log.Data{"config": cfg}, log.INFO)
 
+	// Get HTTP Server with ... ?
 	r := mux.NewRouter()
```

Comment:
if we add generic `ADD CODE` comments, I would suggest adding one here to mention middleware, something like:
```
// ADD CODE: Add any middleware that your service requires
```

Reply:
done, but please check my next commit